templates/frontEndLayout.html

```diff
@@ -289,16 +299,218 @@
+    } else {
+        window.alert("No " + GAME_CONFIG.seeds[seed].name + " seeds to sell.");
+    }
+};
+
+plant = function(x, y, seed) {
+    if (state[seed] > 0) {
+        state[seed] --;
+        document.getElementById("owned" + seed).innerHTML = state[seed];
+        var bigBox = document.getElementById("plot_" + x + "_" + y);
+        var medBoxes = bigBox.getElementsByClassName("container2");
+        for(i = 0; i < medBoxes.length; i++) {
+            medBoxes[i].style.display = "none";
+        }
+        var box = document.getElementById("growing" + x + y);
+        box.style.display = "block";
+        box.style.backgroundImage = "url(" + GAME_CONFIG.seeds[seed].imageLarge + ")";
+        var soil = document.getElementById("soil" + x + y);
+        soil.style.display = "none";
+        state["plot" + x + y].seedType = seed;
+        state["plot" + x + y].timeLeft = GAME_CONFIG.seeds[seed].harvestTimeSeconds;
+        updateDisplay();
+        var progressBar = document.getElementById("progressBar" + x + y);
+        progressBar.style.display = "block";
+    } else {
+        window.alert("No " + GAME_CONFIG.seeds[seed].name + " seeds to plant.");
+    }
+};
+
+harvest = function(x, y) {
+    var seed = state["plot" + x + y].seedType;
+    state[seed] += GAME_CONFIG.seeds[seed].harvestYield;
+    document.getElementById("owned" + seed).innerHTML = state[seed];
+    var bigBox = document.getElementById("plot_" + x + "_" + y);
+    var medBoxes = bigBox.getElementsByClassName("container2");
+    for(i = 0; i < medBoxes.length; i++) {
+        medBoxes[i].style.display = "block";
+    }
+    var box = document.getElementById("growing" + x + y);
+    box.style.display = "none";
+    box.style.backgroundImage = "url(" + GAME_CONFIG.seeds[seed].imageLarge + ")";
+    var soil = document.getElementById("soil" + x + y);
+    soil.style.display = "block";
+    var button = document.getElementById("button" + x + y);
+    button.style.display = "none";
+    seed = 0;
+}
+
+tick = function() {
+    for(i = 0; i <= 2; i++) {
+        for(j = 0; j <= 2; j++) {
+            if(state["plot" + i + j].seedType != 0) {
+                var x = state["plot" + i + j].timeLeft;
+                var y = state["plot" + i + j].seedType;
+                var percent = x / GAME_CONFIG.seeds[y].harvestTimeSeconds
+                document.getElementById("filler" + i + j).style.width = percent * 100 + "%";
+                var clock = Math.floor(x / 60) + ":" + x % 60;
+                document.getElementById("time" + i + j).innerHTML = clock;
+                state["plot" + i + j].timeLeft --;
+                if (x == 0) {
+                    document.getElementById("progressBar" + i + j).style.display = "none";
+                    document.getElementById("button" + i + j).style.display = "block";
+                }
+
+            };
+        };
+    };
+};
+
+updateDisplay = function() {
```

Comment:
this second updateDisplay function works, but there's a lot of duplicate code.

Try refactoring your tick() function to use this new updateDisplay function.  That should make tick much simpler.